ksp_plugin_adapter/manœuvre_marker.cs

```diff
@@ -147,14 +213,39 @@ internal class ManœuvreMarker : UnityEngine.MonoBehaviour {
   }
 
   public void OnMouseUp() {
+    if (!has_dragged_this_click_) {
+      OnMouseClicked();
+    }
     is_dragged = false;
+    has_dragged_this_click_ = false;
+  }
+
+  public void OnMouseClicked() {
+    main_window_.Show();
+    flight_planner_.Show();
+    flight_planner_.RequestEditorFocus(index_);
   }
 
   public void OnMouseExit() {
     is_hovered = false;
   }
 #endregion Mouse Events
 
+  // Accumulate and reset the interactivity states of all active markers.
+  public void Update() {
+    has_interacting_marker |= is_interacting;
+
+    // Only run one copy of this coroutine.
+    if (index_ == 0) {
```

Comment:
This will probably break if the first manœuvre in the plan is 'anomalous'. However, I'm not sure what that means and when that might occur.

Reply:
Manœuvres are not anomalous per se.  Segments are the root cause of anomalies.  A segment (which may be a coast or a burn) is anomalous iff we fail to integrate it, for instance because we reached the maximum number of steps without reaching the desired end time (typically this would happen if the segment goes very close to a singularity).  When that happens, all the manœuvres that follow the first anomalous segment are themselves marked as anomalous.  If a manœuvre is later changed by the user so that integration succeeds, segments (and manœuvres) that were marked as anomalous can become normal again.

Long ago, we used to delete the manœuvres that followed the first segment where integration failed.  This was very obnoxious: say that you prepare a flight plan with 3 manœuvres, you tweak the first one and then the following coast goes close to the centre of a celestial for a second and, lo and behold, you've lost the last 2 manœuvres.  Anomalous segments and manœuvres were added to avoid losing these manœuvres.

Reply:
Thanks. In that case, the above snippet should be sound, as if the first manœuvre is anomalous and not drawn, subsequent manœuvres will not be drawn either.